crates/oxc_linter/src/rules/jest/prefer_to_be.rs

```diff
@@ -0,0 +1,511 @@
+            .modifiers()
+            .iter()
+            .filter(|modifier| modifier.is_name_equal("not"))
+            .count()
+            > 0;
+
+        if has_not_modifier {
+            if matcher.is_name_equal("toBeUndefined") {
+                Self::check_and_fix("Defined", call_expr, matcher, &jest_expect_fn_call, ctx);
+                return;
+            } else if matcher.is_name_equal("toBeDefined") {
+                Self::check_and_fix("Undefined", call_expr, matcher, &jest_expect_fn_call, ctx);
+                return;
+            }
+        }
+
+        if !is_equality_matcher(matcher) || jest_expect_fn_call.args.len() == 0 {
+            return;
+        }
+
+        let first_matcher_arg = get_first_matcher_arg(&jest_expect_fn_call);
+
+        if first_matcher_arg.is_undefined() {
+            let kind = if has_not_modifier { "Defined" } else { "Undefined" };
+            Self::check_and_fix(kind, call_expr, matcher, &jest_expect_fn_call, ctx);
+            return;
+        }
+
+        if first_matcher_arg.is_nan() {
+            Self::check_and_fix("NaN", call_expr, matcher, &jest_expect_fn_call, ctx);
+            return;
+        }
+
+        if first_matcher_arg.is_null() {
+            Self::check_and_fix("Null", call_expr, matcher, &jest_expect_fn_call, ctx);
+            return;
+        }
+
+        if Self::should_use_tobe(first_matcher_arg) && !matcher.is_name_equal("toBe") {
+            Self::check_and_fix("", call_expr, matcher, &jest_expect_fn_call, ctx);
+        }
+    }
+
+    fn should_use_tobe(first_matcher_arg: &Expression) -> bool {
+        let mut expr = first_matcher_arg;
+
+        if let Expression::UnaryExpression(unary_expr) = expr {
+            if unary_expr.operator.as_str() == "-" {
+                expr = &unary_expr.argument;
+            }
+        }
+
+        if matches!(expr, Expression::RegExpLiteral(_)) {
+            return false;
+        }
+
+        matches!(
+            expr,
+            Expression::BigintLiteral(_)
+                | Expression::BooleanLiteral(_)
+                | Expression::NumericLiteral(_)
+                | Expression::NullLiteral(_)
+                | Expression::TemplateLiteral(_)
+                | Expression::StringLiteral(_)
+        )
+    }
+
+    fn check_and_fix(
+        kind: &str,
+        call_expr: &CallExpression,
+        matcher: &KnownMemberExpressionProperty,
+        jest_expect_fn_call: &ParsedExpectFnCall,
+        ctx: &LintContext,
+    ) {
+        let span = matcher.span;
+        let end = call_expr.span.end;
+        let Some(Expression::MemberExpression(mem_expr)) = matcher.parent else {
+            return;
+        };
+        let is_cmp_mem_expr = matches!(mem_expr.0, MemberExpression::ComputedMemberExpression(_));
+        let modifiers = jest_expect_fn_call.modifiers();
+        let maybe_not_modifier = modifiers.iter().find(|modifier| modifier.is_name_equal("not"));
+
+        if kind == "Undefined" {
```

Comment:
Would you like to use `Enum` to replace this string flag? I think it will be more robust.

Reply:
Update.